OloEngine/src/OloEngine/Physics3D/JoltScene.cpp

```
@@ -26,6 +26,18 @@
+    namespace
+    {
+        // JPH::Factory and the RTTI registered by JPH::RegisterTypes() are
+        // process-wide singletons. Each JoltScene owns its own PhysicsSystem
+        // but they all share these globals, so re-creating them on every
+        // JoltScene::Initialize() leaked the previous Factory plus its
+        // string_view/u32 RTTI hash tables whenever two JoltScenes coexisted
+        // (caught by LSan via MultipleScenesCoexistTest). Ref-count instead
+        // so global state is allocated on the first scene and torn down
+        // exactly once when the last scene shuts down.
+        std::atomic<i32> s_JoltGlobalRefCount{ 0 };
+    } // namespace
```

Cppcheck (2.20.0)

[style] The function 'IsInitialized' is never used.

(unusedFunction)